assembly/src/errors.rs

```diff
@@ -71,8 +71,15 @@ impl AssemblyError {
         Self::ExportedProcInProgram(proc_name.to_string())
     }
 
-    pub fn imported_proc_module_not_found(proc_id: &ProcedureId) -> Self {
-        Self::ImportedProcModuleNotFound(*proc_id)
+    pub fn imported_proc_module_not_found(
+        proc_id: &ProcedureId,
+        proc_name: Option<ProcedureName>,
+    ) -> Self {
+        if let Some(proc_name) = proc_name {
+            Self::ImportedProcModuleNotFound(*proc_id, proc_name)
+        } else {
+            Self::ImportedProcModuleNotFound(*proc_id, ProcedureName::try_from("Unknown").unwrap())
```

Comment:
minor nit: let's change the name to "[unknown]". It will appear in the error message better this way.